salt/modules/zypper.py

```diff
@@ -649,10 +649,10 @@ def version_cmp(ver1, ver2, ignore_epoch=False):
     return __salt__['lowpkg.version_cmp'](ver1, ver2, ignore_epoch=ignore_epoch)
 
 
-def list_pkgs(versions_as_list=False, **kwargs):
+def list_pkgs(versions_as_list=False, attr=None, **kwargs):
```

Comment:
nix.py is new in development
```
    :param bool attributes:
        show the attributes of the packages when listing all packages.
        Default: True
```

Which standard is going to be used attributes=True/False  or attrib=[list of attributes]? If its attrib=[list of attributes] then I suggest nix.py should be changed.

At the end of the day their needs to be common parameters which perform the same thing.  Making salt easier to learn and (mostly) consistent across platforms.

Should be able to sit at the command prompt salt * pkg.list_pkgs  param=value
And platforms which do not support param ignore it and platforms which support param do all the same thing.  The attributes return should have the same meaning across platforms. 

Unfortunately and overall/platform wide  doco for pkg.list_pkgs for example does not exist. So it means developers need to look at all of pkg modules to try and determine consistence approach before coding.

Something like `attrib` (common) and `attrib_all`  this would be attributes which are not common across platforms, like the win_*.  Should platform specific attributes be prefixed?

This topic comes up now and then with regards to pkg modules api. Last time, a Salt Team member change it to (x=default,y=default) and then after discussion, the Salt Team changed it back to using kwargs so its more compatible with other platforms.

Configuration Management Tools should mostly remove the platform specific differences so their is common approach across platforms.